src/react-components/room/useObjectList.js

```diff
@@ -0,0 +1,242 @@
+import React, { useState, useEffect, useContext, createContext, useCallback, Children, cloneElement } from "react";
+import PropTypes from "prop-types";
+import { mediaSort, getMediaType } from "../../utils/media-sorting.js";
+
+function getDisplayString(el) {
+  // Having a listed-media component does not guarantee the existence of a media-loader component,
+  // so don't crash if there isn't one.
+  const url = (el.components["media-loader"] && el.components["media-loader"].data.src) || "";
+  const split = url.split("/");
+  const resourceName = split[split.length - 1].split("?")[0];
+  let httpIndex = -1;
+  for (let i = 0; i < split.length; i++) {
+    if (split[i].indexOf("http") !== -1) {
+      httpIndex = i;
+    }
+  }
+
+  let host = "";
+  let lessHost = "";
+  if (httpIndex !== -1 && split.length > httpIndex + 3) {
+    host = split[httpIndex + 2];
+    const hostSplit = host.split(".");
+    if (hostSplit.length > 1) {
+      lessHost = `${hostSplit[hostSplit.length - 2]}.${hostSplit[hostSplit.length - 1]}`;
+    }
+  }
+
+  const firstPart =
+    url.indexOf("poly.google") !== -1
+      ? "Google Poly"
+      : url.indexOf("sketchfab.com") !== -1
+        ? "Sketchfab"
+        : url.indexOf("youtube.com") !== -1
+          ? "YouTube"
+          : lessHost;
+
+  return `${firstPart} ... ${resourceName.substr(0, 4)}`;
+}
+
+const ObjectListContext = createContext({
+  objects: [],
+  focusedObject: null,
+  selectedObject: null,
+  focusObject: () => {},
+  unfocusObject: () => {},
+  inspectObject: () => {},
+  uninspectObject: () => {}
+});
+
+export function ObjectListProvider({ scene, children }) {
+  const [objects, setObjects] = useState([]);
+  const [focusedObject, setFocusedObject] = useState(null); // The object currently shown in the viewport
+  const [selectedObject, setSelectedObject] = useState(null); // The object currently selected in the object list
+
+  useEffect(
+    () => {
+      function updateMediaEntities() {
+        const objects = scene.systems["listed-media"].els.sort(mediaSort).map(el => ({
+          id: el.object3D.id,
+          name: getDisplayString(el),
+          type: getMediaType(el),
+          el,
+          enableLights: false
```

Comment:
As far as I can tell, none of the codepaths change this to true. Previously, there was UI to toggle lights, which would set the "show-background-while-inspecting" flag in local storage, but this PR removes that as well.

In other words, there is no mechanism for a user to enable lights while inspecting an object through the object list. If it's not an omission in the UX, maybe we should remove this state from the `objects` array, and remove references to the local storage flag as well.

Reply:
If you inspect an object using the in-world ui, the lights are enabled, if you inspect an object from the object list then they are not enabled. I didn't want to change that behavior so this is how I implemented that.

Reply:
Ah, I get it now.